File: scripts/setup-momo-api.py
#!/usr/bin/env python3
import requests
import uuid
import json
import base64

# Your subscription key
SUBSCRIPTION_KEY = 'bb5e247b9bc44006bfa82c4aa5d90997'
BASE_URL = 'https://sandbox.momodeveloper.mtn.com'

# Generate a UUID for the API user
reference_id = str(uuid.uuid4())
print(f'Generated Reference ID (X-Reference-Id): {reference_id}')

def create_momo_api_user():
    try:
        print('\n=== Step 1: Creating MoMo API User ===')
        
        # Step 1: Create API User
        headers = {
            'Content-Type': 'application/json',
            'X-Reference-Id': reference_id,
            'Ocp-Apim-Subscription-Key': SUBSCRIPTION_KEY
        }
        
        data = {
            'providerCallbackHost': 'https://webhook.site/unique-id-here'
        }
        
        response = requests.post(
            f'{BASE_URL}/v1_0/apiuser',
            headers=headers,
            json=data
        )
        
        if response.status_code == 201:
            print('✅ API User created successfully!')
        else:
            print(f'❌ Failed to create API user: {response.status_code}')
            print(f'Error details: {response.text}')
            return None

        print('\n=== Step 2: Getting User Information ===')
        
        # Step 2: Get User Information
        headers = {
            'Ocp-Apim-Subscription-Key': SUBSCRIPTION_KEY
        }
        
        response = requests.get(
            f'{BASE_URL}/v1_0/apiuser/{reference_id}',
            headers=headers
        )
        
        if response.status_code == 200:
            user_info = response.json()
            print('✅ User information retrieved:')
            print(f'Target Environment: {user_info.get("targetEnvironment")}')
            print(f'Provider Callback Host: {user_info.get("providerCallbackHost")}')
        else:
            print(f'❌ Failed to get user information: {response.status_code}')
            return None

        print('\n=== Step 3: Creating API Key ===')
        
        # Step 3: Create API Key
        headers = {
            'Ocp-Apim-Subscription-Key': SUBSCRIPTION_KEY
        }
        
        response = requests.post(
            f'{BASE_URL}/v1_0/apiuser/{reference_id}/apikey',
            headers=headers
        )
        
        if response.status_code == 201:
            api_key_data = response.json()
            api_key = api_key_data.get('apiKey')
            print('✅ API Key created successfully!')
            print(f'API Key: {api_key}')
            
            print('\n=== Environment Variables to Update ===')
            print('Add these to your .env.local file:')
            print(f'MOMO_API_USER_ID={reference_id}')
            print(f'MOMO_API_KEY={api_key}')
            
            return {
                'userId': reference_id,
                'apiKey': api_key
            }
        else:
            print(f'❌ Failed to create API key: {response.status_code}')
            print(f'Error details: {response.text}')
            return None

    except Exception as error:
        print(f'❌ Error during API setup: {str(error)}')
        return None

def test_api_access(user_id, api_key):
    try:
        print('\n=== Step 4: Testing API Access ===')
        
        # Create basic auth header
        credentials = f'{user_id}:{api_key}'
        encoded_credentials = base64.b64encode(credentials.encode()).decode()
        
        headers = {
            'Authorization': f'Basic {encoded_credentials}',
            'Ocp-Apim-Subscription-Key': SUBSCRIPTION_KEY,
            'Content-Type': 'application/json'
        }
        
        response = requests.post(
            f'{BASE_URL}/collection/token/',
            headers=headers
        )
        
        if response.status_code == 200:
            token_data = response.json()
            print('✅ Access token retrieved successfully!')
            print(f'Token Type: {token_data.get("token_type")}')
            print(f'Expires In: {token_data.get("expires_in")} seconds')
            print('\n🎉 MoMo API setup completed successfully!')
            print('\nYour API is ready to use for payments!')
        else:
            print(f'❌ Failed to get access token: {response.status_code}')
            print(f'Error details: {response.text}')
            
    except Exception as error:
        print(f'❌ Error testing API access: {str(error)}')

def main():
    print('🚀 Setting up MoMo API credentials...')
    print(f'Subscription Key: {SUBSCRIPTION_KEY}')
    
    credentials = create_momo_api_user()
    
    if credentials:
        test_api_access(credentials['userId'], credentials['apiKey'])
        
        # Write credentials to a file for easy copying
        with open('momo-credentials.txt', 'w') as f:
            f.write(f'MOMO_API_USER_ID={credentials["userId"]}\n')
            f.write(f'MOMO_API_KEY={credentials["apiKey"]}\n')
        
        print(f'\n📝 Credentials saved to momo-credentials.txt')

if __name__ == '__main__':
    main()

File: scripts/setup-momo-api.js
const { v4: uuidv4 } = require('uuid');

// Your subscription key
const SUBSCRIPTION_KEY = 'bb5e247b9bc44006bfa82c4aa5d90997';
const BASE_URL = 'https://sandbox.momodeveloper.mtn.com';

// Generate a UUID for the API user
const referenceId = uuidv4();
console.log('Generated Reference ID (X-Reference-Id):', referenceId);

async function createMoMoApiUser() {
  try {
    console.log('\n=== Step 1: Creating MoMo API User ===');
    
    // Step 1: Create API User
    const createUserResponse = await fetch(`${BASE_URL}/v1_0/apiuser`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'X-Reference-Id': referenceId,
        'Ocp-Apim-Subscription-Key': SUBSCRIPTION_KEY
      },
      body: JSON.stringify({
        providerCallbackHost: 'https://webhook.site/unique-id-here'
      })
    });

    if (createUserResponse.status === 201) {
      console.log('✅ API User created successfully!');
    } else {
      console.log('❌ Failed to create API user:', createUserResponse.status);
      const errorText = await createUserResponse.text();
      console.log('Error details:', errorText);
      return;
    }

    console.log('\n=== Step 2: Getting User Information ===');
    
    // Step 2: Get User Information
    const getUserResponse = await fetch(`${BASE_URL}/v1_0/apiuser/${referenceId}`, {
      method: 'GET',
      headers: {
        'Ocp-Apim-Subscription-Key': SUBSCRIPTION_KEY
      }
    });

    if (getUserResponse.ok) {
      const userInfo = await getUserResponse.json();
      console.log('✅ User information retrieved:');
      console.log('Target Environment:', userInfo.targetEnvironment);
      console.log('Provider Callback Host:', userInfo.providerCallbackHost);
    } else {
      console.log('❌ Failed to get user information:', getUserResponse.status);
      return;
    }

    console.log('\n=== Step 3: Creating API Key ===');
    
    // Step 3: Create API Key
    const createApiKeyResponse = await fetch(`${BASE_URL}/v1_0/apiuser/${referenceId}/apikey`, {
      method: 'POST',
      headers: {
        'Ocp-Apim-Subscription-Key': SUBSCRIPTION_KEY
      }
    });

    if (createApiKeyResponse.ok) {
      const apiKeyData = await createApiKeyResponse.json();
      console.log('✅ API Key created successfully!');
      console.log('API Key:', apiKeyData.apiKey);
      
      console.log('\n=== Environment Variables to Update ===');
      console.log('Add these to your .env.local file:');
      console.log(`MOMO_API_USER_ID=${referenceId}`);
      console.log(`MOMO_API_KEY=${apiKeyData.apiKey}`);
      
      return {
        userId: referenceId,
        apiKey: apiKeyData.apiKey
      };
    } else {
      console.log('❌ Failed to create API key:', createApiKeyResponse.status);
      const errorText = await createApiKeyResponse.text();
      console.log('Error details:', errorText);
    }

  } catch (error) {
    console.error('❌ Error during API setup:', error.message);
  }
}

// Test function to verify the setup
async function testApiAccess(userId, apiKey) {
  try {
    console.log('\n=== Step 4: Testing API Access ===');
    
    // Get access token
    const tokenResponse = await fetch(`${BASE_URL}/collection/token/`, {
      method: 'POST',
      headers: {
        'Authorization': `Basic ${Buffer.from(`${userId}:${apiKey}`).toString('base64')}`,
        'Ocp-Apim-Subscription-Key': SUBSCRIPTION_KEY,
        'Content-Type': 'application/json'
      }
    });

    if (tokenResponse.ok) {
      const tokenData = await tokenResponse.json();
      console.log('✅ Access token retrieved successfully!');
      console.log('Token Type:', tokenData.token_type);
      console.log('Expires In:', tokenData.expires_in, 'seconds');
      console.log('\n🎉 MoMo API setup completed successfully!');
      console.log('\nYour API is ready to use for payments!');
    } else {
      console.log('❌ Failed to get access token:', tokenResponse.status);
      const errorText = await tokenResponse.text();
      console.log('Error details:', errorText);
    }
  } catch (error) {
    console.error('❌ Error testing API access:', error.message);
  }
}

// Main execution
async function main() {
  console.log('🚀 Setting up MoMo API credentials...');
  console.log('Subscription Key:', SUBSCRIPTION_KEY);
  
  const credentials = await createMoMoApiUser();
  
  if (credentials) {
    await testApiAccess(credentials.userId, credentials.apiKey);
  }
}

// Run the setup
main().catch(console.error);